Mono.Addins/Mono.Addins/AddinEngine.cs

```diff
@@ -262,9 +263,23 @@ internal void Initialize (Assembly startupAsm, string customStartupDirectory, st
 		Assembly CurrentDomainAssemblyResolve(object sender, ResolveEventArgs args)
 		{
 			lock (LocalLock) {
-				// MS.NET is more strict than Mono when loading assemblies. Assemblies loaded in the "Load" context can't see assemblies loaded
-				// in the "LoadFrom" context, unless assemblies are explicitly resolved in the AssemblyResolve event.
-				return loadedAddins.Values.Where(a => a.AssembliesLoaded).SelectMany(a => a.Assemblies).FirstOrDefault(a => a.FullName.ToString () == args.Name);
+				string assemblyName = args.Name;
+
+				if (assemblyResolvePaths.TryGetValue (assemblyName, out var inAddin)) {
```

Comment:
I said this on the original PR and I'm still considering it. Maybe each RuntimeAddin registering it's own AssemblyResolve handler is a better option, because we can remove the handler once all of its registered assemblies are loaded.